### PR TITLE
Docstring fix in create_statefbk_iosystem

### PR DESCRIPTION
**Pull request by KybernetikJo**
This PR only changes the docstring of create_statefbk_iosystem, makes help more readable.

- Changes math formulas to sphinx math
- Changes most signal-names to inline math in order to be consistent with formulas
- Fixes some typos
- Adds a full example

### CODE REVIEW

File: control/statefbk.py [hunk 1]
```diff
@@ -589,9 +589,9 @@ def create_statefbk_iosystem(
     This function creates an input/output system that implements a
     state feedback controller of the form
 
-        u = ud - K_p (x - xd) - K_i integral(C x - C x_d)
+    .. math :: u = u_d - K_p (x - x_d) - K_i \int(C x - C x_d)
```
Comment (murrayrm) [inline control/statefbk.py:592]: This is probably OK, but see the general comment regarding keeping equations to a minimum.

File: control/statefbk.py [hunk 2]
```diff
@@ -603,9 +603,9 @@ def create_statefbk_iosystem(
     gains and a corresponding list of values of a set of scheduling
     variables.  In this case, the controller has the form
 
-        u = ud - K_p(mu) (x - xd) - K_i(mu) integral(C x - C x_d)
+    .. math :: u = u_d - K_p(\mu) (x - x_d) - K_i(\mu) \int(C x - C x_d)
 
-    where mu represents the scheduling variable.
+    where :math:`\mu` represents the scheduling variable.
```
Comment (murrayrm) [inline control/statefbk.py:608]: As above, it might make more sense here to just use text rather than LaTeX.

File: control/statefbk.py [hunk 3]
```diff
@@ -614,7 +614,7 @@ def create_statefbk_iosystem(
         is given, the output of this system should represent the full state.
 
     gain : ndarray or tuple
-        If an array is given, it represents the state feedback gain (K).
+        If an array is given, it represents the state feedback gain (`K`).
```
Comment (murrayrm) [inline control/statefbk.py:617]: I would leave this as K to make it easier to read on a terminal.

File: control/statefbk.py [hunk 4]
```diff
@@ -623,18 +623,18 @@ def create_statefbk_iosystem(
 
         If a tuple is given, then it specifies a gain schedule.  The tuple
         should be of the form `(gains, points)` where gains is a list of
-        gains :math:`K_j` and points is a list of values :math:`\\mu_j` at
+        gains :math:`K_j` and points is a list of values :math:`\mu_j` at
```
Comment (murrayrm) [inline control/statefbk.py:626]: I would change this to `mu_j` (get rid of the :math: altogether).

File: control/statefbk.py [hunk 5]
```diff
@@ -623,18 +623,18 @@ def create_statefbk_iosystem(
...
         which the gains are computed.  The `gainsched_indices` parameter
         should be used to specify the scheduling variables.
 
     xd_labels, ud_labels : str or list of str, optional
         Set the name of the signals to use for the desired state and
         inputs.  If a single string is specified, it should be a
         format string using the variable `i` as an index.  Otherwise,
-        a list of strings matching the size of xd and ud,
+        a list of strings matching the size of :math:`x_d` and :math:`u_d`,
```
Comment (murrayrm) [inline control/statefbk.py:634]: Suggest leaving as is.

File: control/statefbk.py [hunk 6]
```diff
@@ -650,13 +650,13 @@ def create_statefbk_iosystem(
...
+        the controller is the desired state :math:`x_d`, the desired input :math:`u_d`, and
+        the system state :math:`x` (or state estimate :math:`\hat{x}`, if an estimator is
         given). If value is an integer `q`, the first `q` values of the
-        [xd, ud, x] vector are used.  Otherwise, the value should be a
+        :math:`[x_d, u_d, x]` vector are used.  Otherwise, the value should be a
         slice or a list of indices.  The list of indices can be specified
-        as either integer offsets or as signal names.  The default is to
-        use the desired state xd.
+        as either integer offsets or as signal names. The default is to
+        use the desired state :math:`x_d`.
```
Comment (murrayrm) [inline control/statefbk.py:659]: I would leave all of this unchanged.

File: control/statefbk.py [hunk 7]
```diff
@@ -677,9 +677,9 @@ def create_statefbk_iosystem(
     -------
     ctrl : NonlinearIOSystem
         Input/output system representing the controller.  This system
-        takes as inputs the desired state `xd`, the desired input
-        `ud`, and either the system state `x` or the estimated state
-        `xhat`.  It outputs the controller action `u` according to the
+        takes as inputs the desired state :math:`x_d`, the desired input
+        :math:`u_d`, and either the system state :math:`x` or the estimated state
+        :math:`\hat{x}`.  It outputs the controller action :math:`u` according to the
         formula :math:`u = u_d - K(x - x_d)`.  If the keyword
```
Comment (murrayrm) [inline control/statefbk.py:683]: I would leave this unchanged and convert the formula on line 683 to remove the :math: directive (so `u = ud - K(x - xd)`).

File: control/statefbk.py [hunk 8]
```diff
@@ -690,8 +690,8 @@ def create_statefbk_iosystem(
 
     clsys : NonlinearIOSystem
         Input/output system representing the closed loop system.  This
-        systems takes as inputs the desired trajectory `(xd, ud)` and
-        outputs the system state `x` and the applied input `u`
+        system takes as inputs the desired trajectory :math:`(x_d, u_d)` and
+        outputs the system state :math:`x` and the applied input :math:`u`
```
Comment (murrayrm) [inline control/statefbk.py:694]: I would remove the :math: directives.

**Review by murrayrm — COMMENTED** [review]
The numpydoc style guide says:
> Equations : as discussed in the [Notes](https://numpydoc.readthedocs.io/en/latest/format.html#notes) section above, LaTeX formatting should be kept to a minimum. Often it’s possible to show equations as Python code or pseudo-code instead, which is much more readable in a terminal. 

Based on this, I have made several suggestions below regarding reverting to plain text in the docstrings, for better display on terminals.

@sawyerbfuller @bnavigator Any thoughts on this general style issue?  If we think that Sphinx/readthedocs is the main way people will digest documentation, I am OK with using the :math: directive more liberally.

**Comment by bnavigator** [comment]
> Any thoughts on this general style issue? If we think that Sphinx/readthedocs is the main way people will digest documentation, I am OK with using the :math: directive more liberally.

I agree with all your comments there.

More complex equations should be set in math and most users will

   * use some sort of Sphinx (RTD or inline help in IDEs like Spyder, JupyterLab might get it in the future too: https://github.com/jupyterlab/jupyterlab/issues/8581), or
   * will understand to read LaTeX, having control systems scientists as our main audience.

OTOH we should not go overboard for single symbols.

File: control/statefbk.py [hunk 9]
```diff
@@ -721,6 +721,23 @@ def create_statefbk_iosystem(
...
+    >>> A = [[0, 1], [-0.5, -0.1]]
+    >>> B = [[0], [1]]
+    >>> C = np.eye(2)
+    >>> D = np.zeros((2, 1))
+    >>> sys = ct.ss(A, B, C, D)
+    >>>
+    >>> Q = np.eye(2)
+    >>> R = np.eye(1)
+    >>>
+    >>> K, _, _ = ct.lqr(sys,Q,R)
```
Comment (bnavigator) [inline control/statefbk.py:738]:
```suggestion
    >>> K, _, _ = ct.lqr(sys, Q, R)
```

File: control/statefbk.py [hunk 10]
```diff
@@ -721,6 +721,23 @@ def create_statefbk_iosystem(
         System name. If unspecified, a generic name <sys[id]> is generated
         with a unique integer id.
 
+    Examples
+    --------
+    >>> import control as ct
+    >>> import numpy as np
+    >>>
```
Comment (bnavigator) [inline control/statefbk.py:728]: `np` and `ct` are globally implicit

https://github.com/python-control/python-control/blob/42c6fb1498aa9f5d542113da288c7c4e3547c117/doc/conf.py#L280-L282

**Comment by KybernetikJo** [comment]
> The numpydoc style guide says:
> 
> > Equations : as discussed in the [Notes](https://numpydoc.readthedocs.io/en/latest/format.html#notes) section above, LaTeX formatting should be kept to a minimum. Often it’s possible to show equations as Python code or pseudo-code instead, which is much more readable in a terminal.
> 
> Based on this, I have made several suggestions below regarding reverting to plain text in the docstrings, for better display on terminals.
> 
> @sawyerbfuller @bnavigator Any thoughts on this general style issue? If we think that Sphinx/readthedocs is the main way people will digest documentation, I am OK with using the :math: directive more liberally.

I was completely unaware of that part of numpydoc, and I'm ok with us not merging it.
(It was a good remark, because I would like to change Slycot/slycot/analysis.py to numpydoc. I can take that into account over there.)

What should we do with this PR?
- Close this PR with comment.
- Keep it open, not do anything, just keep it as reference. Maybe latex rendering in terminals will change at some point in the future.
- Incorporate comments from @murrayrm, @bnavigator and merge. I am not sure it worth it, though.

I am fine with all three variants.

**Comment by bnavigator** [comment]
Although there is little left after incorporating our comments, I think even minor typo fixes and formatting are worth merging. Keep on working and thanks for your contributions!

**Pull request merged**